cloudferrylib/utils/files.py

```diff
@@ -68,7 +68,10 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.runner.run_ignoring_errors(_unlink_dir(self.dirname))
-        return self
+        if exc_type is None:
+            return self
+        else:
+            raise exc_type
```

Comment:
Would be better to simply delete `return` statement: it would re-raise exception and keep all the necessary exception information untouched, while in your solution you're hiding `exc_val` and `exc_tb`.